opentelemetry-sdk/src/opentelemetry/sdk/_metrics/aggregation.py

```diff
@@ -53,57 +47,56 @@ def collect(self) -> Optional[_PointVarT]:
         pass
 
 
-class SynchronousSumAggregation(
-    _InstrumentMonotonicityAwareAggregation, Aggregation[Sum]
-):
-    def __init__(self, instrument_is_monotonic: bool):
-        super().__init__(instrument_is_monotonic)
-        self._value = 0
+class SumAggregation(Aggregation[Sum]):
+    def __init__(
+        self,
+        instrument_is_monotonic: bool,
+        instrument_temporality: AggregationTemporality,
+    ):
+        super().__init__()
+        self._instrument_is_monotonic = instrument_is_monotonic
+
+        if instrument_temporality is AggregationTemporality.DELTA:
+            self._value = 0
+
+        else:
+            self._value = None
+
         self._start_time_unix_nano = _time_ns()
+        self._instrument_temporality = instrument_temporality
 
     def aggregate(self, measurement: Measurement) -> None:
-        with self._lock:
-            self._value = self._value + measurement.value
+
+        if self._instrument_temporality is AggregationTemporality.DELTA:
```

Comment:
I might be misunderstanding but shouldn't this be `AggregationTemporality.CUMULATIVE`?

Reply:
Here `self._value` is always a cumulative, right?. If the instrument is already producing cumulative values why would we add again?

Reply:
Is it the instrument that is recording the one that is CUMULATIVE or is it the aggregation? My understanding was that the aggregation is the one with the temporality, so in order to have that behaviour, we have different addition logic. The values that are coming in from instruments are just individual measurements, we don't know anything about them. It is once is reaches the aggregations in which we decide how they are interpreted. So if the aggregation were to be cumulative, to keep it cumulative, the measurement value must be added onto the previous value to keep the TOTAL sum (because we are accumulating). For delta, the new measurement value replaces the old value that we were keeping track of. Is this correct or is my understanding completely wrong?

Reply:
Hmmm, aren't sync counter instruments are reporting the delta values via `request_counter.add(1, ...)` and as [noted](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md#asynchronous-counter-creation) in the spec async counter instruments always report the absolute value and we take the difference b/w successive records to get the rate of change?

Reply:
Right, so that means instruments themselves have assumed temporality (from the spec the sync counters are DELTA and async counters are CUMULATIVE). This doesn't change what the change I proposed above right? The temporality of the aggregation should make it so that `collect` sums up the values if cumulative and reports the value if delta. I'm not too sure about the relationship of that with the assumed temporality of the instruments. Unless I'm misunderstanding something here. @aabmass any input on this?

Reply:
I actually don't think this check is needed at all, and `aggregate()` should behave the same way regardless of instrument temporality. An aggregation can see more than one async measurement in a single interval if you set [`attribute keys`](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/sdk.md#:~:text=A%20list%20of-,attribute%20keys,-(optional).%20If%20provided) on the view, so they have to be summed and not just replaced. There is a [section in the supplementary guidelines showing the desired behavior](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/supplementary-guidelines.md#asynchronous-example-attribute-removal-in-a-view).

I think they only need to be handled differently in `collect()`, where an Aggregation for an async instrument should set temporality to `CUMULATIVE` instead of `DELTA`. To summarize with Sum as an example:

- SumAggregation for view on a ObservableCounter receives several `Measurement`s
  - **It adds them together**
  - When `collect()` is called, it outputs a point with `aggregation_temporality=CUMULATIVE`
  - Next, the temporality conversion process (#2380) will either
    - Return that cumulative if the output temporality is cumulative
    - Subtract the previous cumulative value from it if output temporality is delta.
- SumAggregation for view on a synchronous Counter receives several `Measurement`s
  - **It adds them together**
  - When `collect()` is called, it outputs a point with `aggregation_temporality=DELTA`
  - Next, the temporality conversion process will either
    - Add it to the previous cumulative value if the output temporality is cumulative
    - Return that delta if the output temporality is delta

Reply:
Right this makes sense to me. It's not the `Aggregation` itself that has the temporality but the input instruments that created the measurements and the type of output desired.

Reply:
That makes sense, I have removed the check for `aggregate`. But I have some doubts regarding this:

> Right this makes sense to me. It's not the `Aggregation` itself that has the temporality but the input instruments that created the measurements and the type of output desired.

Isn't the aggregation the one who has the temporality? The temporality is known as aggregation temporality, right? :shrug: Are we really concerned with the instrument temporality or with its _synchronicity_ (it being synchronous or asynchronous)? It seems like we associate synchronous with delta and asynchronous with cumulative, which in that case makes it possible for us to use the instrument temporality to make the decision in collect. The spec seems to be the one making this association, we don't expect to have a delta asynchronous instrument or a cumulative synchronous instrument, right? Because if that was the case, maybe making the decision based on synchronicity would be better? :shrug:

Reply:
Yes that is a good question. I personally am not too sure about that. For now, I say we move ahead with this PR and I'll add this to the list of discussion topics for this weeks SIG.

Reply:
Decided in the SIG: We assume that the user will be using instruments correctly under the assumption that sync instruments have temporality DELTA and async instruments have temporality CUMULATIVE.